### DISCUSSION TITLE
Sonof ZBdongle-P to SLZB-06 migration issue.

### DISCUSSION THREAD

**Discussion by tomtom13** [Koenkk/zigbee2mqtt#19802] [Devices]
Hi, 
I've tired to follow the howto's on the both Z2B and SLZB-06 pages and:
- I'm running Z2B inside of HASS, inside of VM on top of proxmox - installation is pretty much box standard, however with over 150 devices on zigbee network and a availability check enables, set to advanced and timeout of 10 minutes (some dumb battery powered devices can vanish due to low battery while to giving final battery report) 
- I've flashed the SLZB-06 with firmware AND changed the IEEE address to match the Sonoff IEEE address. 
- I've shut down the Z2B
- Unplugged the sonof usb dongle 
- powered up SLZB-06
- altered the configuration to point at the right IP and port
- spun the Z2B back up 
- enabled joining permanently.
- powered down most of the routers (god bless the fuse box), while VM was up and running ( god bless the UPS ;D )

The results is that most of the devices seem to be connected (but in a weird state). About 30 devices don't even want to respond to ping so show up as unavailable. I've waited about an hour for network to stabilise but with no cigar.
Battery operated devices seem to be connected and reporting their state, for example:
- temperature and humidity sensors (tuya) keep pumping out updates
- philips hue motion sensors are pumping out motion events
- philips hue bulbs keep reporting their state (on + colour)
- philips (and frient) power sockets keep reporting their state (and power for frient)

However I'm not able to modify settings / state of any of that device. I can't change the "presence interval" on motion sensor, can't switch the bulb ON/OFF, can't toggle power outlets etc etc.

After that experiment I've unplug the SLZB-06, reconected sonof usb dongle and punched back old configuration. Everything spin up back to life within 2 minutes, with no glitshes (well, 30 seconds, but one bulb took 2 minutes to find it's way back into the heard, I might point out that I wasn't playing with it what so ever) without power cycle.

I presume I did something dumb / skipped a step ... please enlighten me as I would prefer to have a network attached dongle to allow easier failover / more optimal location for it.

Just something that I found in a log (log not present anymore):
`{"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":true,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'MAC no ack' (233)))\`

**Comment by Dinth** [comment]
@tomtom13 Hej, have you managed to resolve that? I will be doing the same thing tomorrow and i was wondering how stable is the process.

Also, how did you change the IEEE of SLZB-06?

**Reply by tomtom13** [reply]
Sorry bud, no luck. I've asked that question in few places and since nobody is actually willing to provide even a hint of help, I'm SOL. At this point I've suspended the effort to get to a point where I can suspend the whole home for a day to perform the shift. 
About the IEEE - as far as I remember I just followed the guide on how to force IEEE address in config (through the HASS page). 
IF you have some luck, please let me know how it went and what you've did.

**Reply by Dinth** [reply]
Hi!
Actually, i have managed to do the migration yesterday. Not without a hiccup, but eventually the mesh has stabilized enough that i can consider it usable (except of a few odd devices). 
IEEE was the first problem i could only change it successfully via TI official app, the problem was that this app did not accept the official firmware (.hex format) so i needed to convert the firmware to .bin format first. 
After the migration, almost all the devices got picked up straight away and then... another hiccup has started: https://github.com/Koenkk/zigbee2mqtt/discussions/20927
But since then i have re-paired a few problematic devices (mainly Ikea bulbs) and things got gradually better. 

Now im only getting those: ```

warning 2024-01-22 13:34:35Failed to ping 'Private office light 2' (attempt 1/1, Read 0x000b57fffe23843b/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":true,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 24703 - 1 - 197 - 0 - 1 after 10000ms))
warning 2024-01-22 13:34:40Failed to ping 'Zigbee repeater first floor' (attempt 1/1, Read 0x00124b002447f894/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":true,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'MAC no ack' (233)))
warning 2024-01-22 13:34:49Failed to ping 'Zigbee repeater ground floor' (attempt 1/1, Read 0x00124b002447f8d4/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":true,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'No network route' (205)))
warning 2024-01-22 13:34:56Failed to ping 'Repeater Master bedroom' (attempt 1/1, Read 0xe0798dfffecfab37/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":true,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'No network route' (205)))
warning 2024-01-22 13:44:49Failed to ping 'Private office light 2' (attempt 1/1, Read 0x000b57fffe23843b/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":true,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 24703 - 1 - 234 - 0 - 1 after 10000ms))
warning 2024-01-22 13:44:55Failed to ping 'Zigbee repeater first floor' (attempt 1/1, Read 0x00124b002447f894/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":true,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'MAC no ack' (233)))
warning 2024-01-22 13:45:03Failed to ping 'Zigbee repeater ground floor' (attempt 1/1, Read 0x00124b002447f8d4/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":true,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'No network route' (205)))
warning 2024-01-22 13:45:10Failed to ping 'Repeater Master bedroom' (attempt 1/1, Read 0xe0798dfffecfab37/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":true,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'No network route' (205)))
```
but only one light bulb and some repeaters (which are no needed anymore as they've been replaced with ZBDongles)

**Reply by tomtom13** [reply]
Yeah, now I remember - I also used the app to change the IEEE address on the device. 
I've had the same messages. Thou, for me - some devices worked without problem, some reported the state just fine but not allowed any control. It's been a while since November and my memory is faint :/ I think that error that you see might be a root case of major issues that I've had.

Now, can you elaborate on "i have re-paired a few problematic devices" please ? Specifically what steps you took. I'm curious, because for me a full removal and re-adding with the same name will cause issues with automations.

**Reply by tomtom13** [reply]
Another thing - I've used the app provided by manufacturer, the Mac version. It "seemed" to work, ie: I've programmed the firmware and IEEE address -> unplugged the device and shut down the programmer -> reconnected the device and fired the programmer back up and read the IEEE address ... it seemed to remember the address that I've programmed, YET from my software engineering experience: "just you programmed a different address according to some glorified GUI, doesn't mean that the new address is going to be used in reality". So where I'm driving it, is whenever you actually had some other form of confirmation that the dongle address changed ?

**Reply by Dinth** [reply]
Yeah, when i click Settings -> Raw in Z2M i can see 
```
    "coordinator": {
        "ieee_address": "0x00124b00
```

and thats the correct address (belonging to the old adapter) 

>Thou, for me - some devices worked without problem, some reported the state just fine but not allowed any control. 

For me some worked fine, some worked but were frequently dropping off for a couple of seconds and some took like 6 second to respond to commands. But i havent tested all of them. 

>Now, can you elaborate on "i have re-paired a few problematic devices" please ?
TBH in the past when i needed a re-pair i rarely needed to remove the device - usually i just do "Permit join" in Z2M and put the device in pairing mode - this should reinterview the node, etc.
And this is what i have done here, enabled permit join, went through my ikea bulbs (those are the first models so always have some issues with them), reset them by switching on and off and observed in Z2M if they are getting reinterviewd.

**Reply by tomtom13** [reply]
what's your mesh size ? my is 155 at the moment.

**Comment by felixbouleau** [comment]
I'm having the same issues. I successfully set the IEEE address (confirmed both by reading the firmware, and from Z2M itself). 

When using the new adapter, I get bombarded with timeout messages. None of my devices are responding to e.g. on/off commands, even after re-pairing them. I'm going to give the mesh another few hours to heal, but I'm not hopeful.

**Reply by Dinth** [reply]
Just a basic troubleshooting question, but have you unplugged the original adapter? 

Also try restarting the SLZB-06 Zigbee router, ESP32 and then restartt Z2M

**Reply by ddeconin-gh** [reply]
In my case, I got the migration working by taking the power off my router/repeater devices after the migration for about 20 seconds. After powering up they found the new router just fine. 
I also used a manual backup of my old coordinator to load into the new one.

**Reply by felixbouleau** [reply]
I ended up just setting up the new adapter as a parallel instance and spending an hour moving my 50 devices over, one by one. It feels nice to start fresh. 

For anyone that comes after me: I tried all of the above *except* cutting power to the old adapter. 🤦  That seems like a very likely culprit. I don't know what I was thinking.

**Reply by Dinth** [reply]
Yeah, you NEED to disconnect the old device, otherwise there will be two devices with same IEEE in the same network.
Im wondering, because i have changed the (secondary) IEEE of the old device, but after that i connected it as a router - can this be a reason of my problems?

**Comment by tomtom13** [comment]
BTW guys, a fun fact ... don't keep those dongles disconnected from power for longer period of time. I've just returned to the project of switching my USB dongle to ethernet dongle and ... it doesn't work .... doesn't even take any power from USB-C port. And I have two dongles, so I took a fresh one out of the box and after 5 months shelf time both are dead.

**Reply by Dinth** [reply]
Doesnt really sound "fun". Have you tried flashing it?

**Reply by tomtom13** [reply]
actually I've managed to resurrect it (at least one so far).
Trick I had to perform was to have it connected to ethernet and provide power out of power bank. Now it works pretty much OK - in terms that it connects to network, works from any power supply, even powers it self up while connected to the laptop. I will report later on whenever I can flash it, or whenever radio work - but definitely it's something to watch out for.

**Reply by tomtom13** [reply]
Soooo I've found what seemed to be an issue - after long period of storage it required a cable that had no data lanes - a dumb cable with just power to kick start it, however the same cable that worked for me to actually resurrect it made my chase my own tail for day or two because it had no data lines and I was unable to upgrade it. 
Firmware upgrade works even on MAC, but one has to use the usbc-usbA -> usbA 2.0-usbc combo to get communication up and running. Also the "wire mod" is required to enable the device enter the update mode, and on the web interface one has to switch it into the USB mode.
Hope that helps anyone.
